crates/tx3-lang/src/lowering.rs

```diff
@@ -410,6 +411,17 @@ impl IntoLower for ast::ListConstructor {
     }
 }
 
+impl IntoLower for ast::TupleConstructor {
+    type Output = ir::Expression;
+
+    fn into_lower(&self, ctx: &Context) -> Result<Self::Output, Error> {
+        let fst = self.fst.into_lower(ctx)?;
+        let snd = self.snd.into_lower(ctx)?;
+
+        Ok(ir::Expression::Tuple(Box::new((fst, snd))))
```

Comment:
Here, instead of lowering into an `ir::Expression:Tuple`, we could just go for a List.